### DISCUSSION TITLE
Grain boundary smoothing

### DISCUSSION THREAD

**Discussion by mardilos** [mtex-toolbox/mtex#1796] [Ask Anything]
Hi all, just started using MTEX for EBSD analysis.
Went through the tutorial video and IPF mapping works fine. I just have problems with the grain boundary smoothing. 
I implemented it the same as in the tutorial:

%% Smooth grain boundaries
grains = smooth(grains,2);

It doesn`t work and i get the following error message:

Error using  * 
Both logical arguments must be scalar. Use TIMES (.*) for elementwise multiplication.

Error in grain2d/smooth (line 26)
A_V = I_VF * I_VF';

Error in EBSD_Data_analysis (line 62)
grains = smooth(grains,2);

Is this a known bug and does anybody know how to fix it?

Thanks 
mardilos

**Comment by lfgmorales** [comment]
Hi

Would be helpful to know which version of MTEX you are using, and what have you done before the grain smoothing, I cannot reproduce the error with version 5.9


cheers


Luiz

**Reply by mardilos** [reply]
Hi Luiz,
thank you for your answer

i am also using version 5.9.0

Before, i am just doing the import routine and the grain calculations (see attached the code and the corresponding test data)
[EBSD_Data_analysis.zip](https://github.com/mtex-toolbox/mtex/files/11284616/EBSD_Data_analysis.zip)
[Au_O-ZnO_Site2_fit.zip](https://github.com/mtex-toolbox/mtex/files/11284624/Au_O-ZnO_Site2_fit.zip)

**Comment by lfgmorales** [comment]
Hi


Maybe try to reinstall and install again and see if you run into errors, because I have no problem to calculate with your data


cheers


Luiz

**Reply by mardilos** [reply]
tried that already
can it also depend on the matlab version itself? it is from 2017...

**Comment by lfgmorales** [comment]
In principle it should be compatible, but if you have the chance to get a newer version, that may solve your problem. Unfortunately I don’t have 2017 Matlab version anymore

You can also try a previous version, like 5.8 or 5.7 and see if that works

cheers

L

**Reply by mardilos** [reply]
Hi Luiz,

so thanks a lot
it works with an earlier version

**Comment by lfgmorales** [comment]
No worries, I just tested 5.9 with mtex 2021a and it also works, so K assume is a matlab incompatibility

Cheers

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 20 Apr 2023, at 16:45, mardilos ***@***.***> wrote:

﻿

Hi Luiz,

so thanks a lot
it works with an earlier version

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1796#discussioncomment-5674740>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN645X4XWZY6GBH5IWTXCFDVHANCNFSM6AAAAAAXDYOCWU>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1796#discussioncomment-5674740", "url": "https://github.com/mtex-toolbox/mtex/discussions/1796#discussioncomment-5674740", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by wqhit307** [comment]
Thanks for asking the question. Same problem just happened to me. And I am the same, using MATLAB 2017b with MTEX5.10. I wish there could be some solution to this issue because updating MATLAB on a company computer is not always straightforward.
